### PR TITLE
[csharp] update TFM to net6 and net8

### LINKED ISSUE #770: [Enhancement] 请RocketMQ dotnet客户端升级到新版
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Programming Language of the Client

C#

### Summary

我本地的代码是.net 8.0，但是客户端并不支持，原因是当前客户端中引用的包，版本过低，与.NET 8.0中默认的包版本有冲突，导致无法正常安装该客户端，请尽快升级一下
image

### Motivation

现在.net 5.0已经过了维护期，很多其他MQ客户端已经至少支持到了.net 6.0和.net 8.0

### Describe the Solution You'd Like

请升级到至少支持.NET 6.0和.NET 8.0

### Describe Alternatives You've Considered

-

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by catcherwong**
### Which Issue(s) This PR Fixes

Fixes #770

### CODE REVIEW

**Comment by catcherwong** [comment]
I don't know why the build failed about macos-11, and no log for it.

**Comment by lizhanhui** [comment]
> I don't know why the build failed about macos-11, and no log for it.

Looks macos-11 is no longer available in public runners. Removed it from build matrix.

File: csharp/rocketmq-client-csharp/Client.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@
 using System.Linq;
 using Microsoft.Extensions.Logging;
 using Proto = Apache.Rocketmq.V2;
-using grpc = Grpc.Core;
+using grpcLib = Grpc.Core;
```
Comment (lizhanhui) [inline csharp/rocketmq-client-csharp/Client.cs:26]: Why rename this alias?

Comment (catcherwong) [reply csharp/rocketmq-client-csharp/Client.cs:26]: Fix ` warning CS8981: The type name
'grpc' only contains lower-cased ascii characters. Such names may become reserved for the language.` after upgrading the version.

**Comment by catcherwong** [comment]
> Looks macos-11 is no longer available in public runners. Removed it from build matrix.

Based on https://github.blog/changelog/2024-05-20-actions-upcoming-changes-to-github-hosted-macos-runners/

I update from macos-11 to macos-12, the ci runs well now.

**Review by lizhanhui — APPROVED** [review]
OK

**Pull request merged**